### ISSUE TITLE
Sequential.eval() does not put model into eval mode

### ISSUE THREAD

**Issue by brianberns** [dotnet/TorchSharp#1426]
Calling `eval()` should make `training` false. However, this does not work for `Sequential` modules.

Example F# program:

```fsharp
open type TorchSharp.torch.nn

let linear = Linear(10, 10)
linear.eval()
assert(not linear.training)       // succeeds

let sequential = Sequential(Linear(10, 10))
sequential.eval()
assert(not sequential.training)   // fails
```

**Comment by yueyinqiu** [comment]
It may not be related to this issue, but actually I suppose we shall reconsider the the submodules, especially the way we register them.

We have discussed this before. https://github.com/dotnet/TorchSharp/issues/1272#issuecomment-2005623635

I think the best solution would be to use source generators. But it's true adding complexity, so we consider it as a last resort before. https://github.com/dotnet/TorchSharp/issues/1272#issuecomment-2010432172

**Comment by alinpahontu2912** [comment]
Hey @brianberns, thanks for the issue. I tested it myself and you are right, there seems to be a problem with the Sequential module. For the moment, would you try creating and using your own custom module containing the Sequential module as specified in the [wiki](https://github.com/dotnet/TorchSharp/wiki/Creating-Your-Own-TorchSharp-Modules) here ? Meaning something like this: 
```
public class CustomModel : Module<Tensor, Tensor>
{
    private readonly Module<Tensor, Tensor> layers;
    public CustomModel()
        : base("CustomModel")
    {

        var modules = new List<(string, Module<Tensor, Tensor>)>();

        modules.Add(("lin1", Linear(10, 10)));

        layers = Sequential(modules);

        RegisterComponents();
    }

    public override Tensor forward(Tensor input)
    {
        return layers.forward(input);
    }

    protected override void Dispose(bool disposing)
    {
        if (disposing)
        {
            layers.Dispose();
        }
        base.Dispose(disposing);
    }
}
